### PR TITLE
Rhamp 3705 sonar

### PR DESCRIPTION
**Pull request by lgriffin**
Addressed wrong source and test dir

### CODE REVIEW

**Comment by lgriffin** [comment]
retest this please

**Comment by lgriffin** [comment]
retest this please

**Comment by lgriffin** [comment]
retest this please

**Comment by grdryn** [comment]
@lgriffin I see in the console output of the sonar job:

```
10:51:38.085 WARN  - No coverage information will be saved because LCOV file cannot be found. Provided LCOV file path: ./coverage/lcov.info
10:51:38.085 WARN  - Provided LCOV file path: ./coverage/lcov.info. Seek file with path: /var/lib/jenkins/workspace/fh-cluster_sonar-pullrequests/./coverage/lcov.info
```

Should the coverage have run first? Do we lose anything by not running the coverage for these PR jobs?

**Comment by lgriffin** [comment]
Hey @grdryn we have fh:default running on this particular setup, unsure how long that has been the default. However there is a wider discussion ongoing about the merits of running fh:coverage on a PR. It really slows down builds like NGUI and we don't gain anything from it as we capture the coverage metrics on the nightly + we can have quality gates to show if we are slacking on the coverage. @wtrocki and @wei-lee  were discussing this previously.

**Comment by grdryn** [comment]
@lgriffin ok, was just checking. :) Should we remove fh:coverage in jobs that still have it then, so that we're not running it needlessly? @wei-lee @wtrocki any opinion there?

**Comment by wtrocki** [comment]
TL;DR Coverage was already removed in most of the jobs by change in `grunt-fh-build`.
Problem was ngui which was different, but I think that we removed that as well. 

---- 
I think that we agreed that list of required stuff will be included in `fh:default`. So for every component running `fh:default` will be ok (apart from ngui which is different). Running coverage is killing our productivity. ngui build takes 12 minutes instead of 4. I think that after we will finish general work with sonar we should review general changes done in jenkins and make sure they are ok and reflected in builder. 

I think that everyone will agree that:
- Coverage should be done on nightly jobs only.
- We should run jshint, unit and integration tests as part of the PR build
- Sonar PR build should only deal with sonar issues - no tests cases there. 

Maybe it makes more sense to move jshint to sonar PR job?

**Comment by grdryn** [comment]
@wtrocki Sounds good to me. I'm happy to leave coverage to the nightly job :)
